pilot/test/util/kubernetes.go

```diff
@@ -210,18 +209,3 @@ func FetchLogs(cl kubernetes.Interface, name, namespace string, container string
 	}
 	return string(raw)
 }
-
-// Eventually retries until f() returns true, or it times out in error
-func Eventually(name string, f func() bool, t *testing.T) {
-	t.Helper()
-	interval := 64 * time.Millisecond
```

Comment:
Note that by default my new Eventually impl fails after 5 seconds, and this function will wait up to 1 minute. I'm curious to see what the tests do with this PR.

Reply:
How about something in the middle - like 25 seconds ? I'm trying to switch default refresh to 10 seconds (due to load), this will give 2 refresh intervals.

Reply:
for tests we shouldn't need to wait 10s for changes to propagate - we don't have that many services that 1s shouldn't work - so we can get through tests faster

Reply:
Made the default 1s poll, 30 second deadline. Probably too generous, but since the goal is to fix flakes, maybe not the worst thing ever.